plugins/database/redshift/redshift_test.go

```diff
@@ -331,79 +383,73 @@ func TestPostgresSQL_SetCredentials(t *testing.T) {
 		t.Fatal(err)
 	}
 	dbUser := "vaultstatictest-" + fmt.Sprintf("%s", uid)
-	createTestPGUser(t, url, dbUser, "1Password", testRoleStaticCreate)
+	createTestPGUser(t, connURL, dbUser, "1Password", testRoleStaticCreate)
 
 	db := newRedshift(true)
 	_, err = db.Init(context.Background(), connectionDetails, true)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	password, err := db.GenerateCredentials(context.Background())
-	if err != nil {
-		t.Fatal(err)
-	}
+	const password1 = "MyTemporaryRootPassword1!"
+	const password2 = "MyTemporaryRootPassword2!"
 
-	usernameConfig := dbplugin.StaticUserConfig{
+	updateReq := newdbplugin.UpdateUserRequest{
 		Username: dbUser,
-		Password: password,
+		Password: &newdbplugin.ChangePassword{
+			NewPassword: password1,
+			Statements:  newdbplugin.Statements{},
+		},
 	}
 
 	// Test with no configured Rotation Statement
-	username, password, err := db.SetCredentials(context.Background(), dbplugin.Statements{}, usernameConfig)
-	if err == nil {
-		t.Fatalf("err: %s", err)
-	}
-
-	statements := dbplugin.Statements{
-		Rotation: []string{testRedshiftStaticRoleRotate},
-	}
-	// User should not exist, make sure we can create
-	username, password, err = db.SetCredentials(context.Background(), statements, usernameConfig)
+	_, err = db.UpdateUser(context.Background(), updateReq)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	if err := testCredsExist(t, url, username, password); err != nil {
+	if err := testCredsExist(t, url, dbUser, password1); err != nil {
 		t.Fatalf("Could not connect with new credentials: %s", err)
 	}
 
-	// call SetCredentials again, password will change
-	newPassword, _ := db.GenerateCredentials(context.Background())
-	usernameConfig.Password = newPassword
-	username, password, err = db.SetCredentials(context.Background(), statements, usernameConfig)
+	// Test with explicitly set rotation statement
+	updateReq = newdbplugin.UpdateUserRequest{
+		Username: dbUser,
+		Password: &newdbplugin.ChangePassword{
+			NewPassword: password2,
+			Statements: newdbplugin.Statements{
+				Commands: []string{testRedshiftStaticRoleRotate},
+			},
+		},
+	}
+	_, err = db.UpdateUser(context.Background(), updateReq)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	if password != newPassword {
-		t.Fatal("passwords should have changed")
-	}
-
-	if err := testCredsExist(t, url, username, password); err != nil {
+	if err := testCredsExist(t, url, dbUser, password2); err != nil {
 		t.Fatalf("Could not connect with new credentials: %s", err)
 	}
 }
 
-func TestPostgreSQL_RotateRootCredentials(t *testing.T) {
-	/*
-		   Extra precaution is taken for rotating root creds because it's assumed that this
-		   test will run against a live redshift cluster. This test must run last because
-		   it is destructive.
+func TestRedshift_UpdateUser_RootPassword(t *testing.T) {
```

Comment:
This test is arguably not valuable anymore, it's kind of a special case of the test above, `TestRedshift_UpdateUser_Password `.

Reply:
I agree - the DB instance doesn't care about which user its updating and doesn't make any changes to its state as a part of this change. I think we can get rid of this test.